app/controllers/users_controller.rb

```diff
@@ -6,4 +6,17 @@ def show
   def index
     @users = User.all
   end
+
+  def update
+    @user = User.find(params[:id])
+    if current_user.authority == "admin"
+      if @user.authority == "user"
+        @user.authority = "mod"
+      elsif @user.authority == "mod"
+        @user.authority = "user"
+      end
```

Comment:
Rather than toggling between roles it might be better to pass the new role in as a parameter:

```ruby
def update
  @user = User.find(params[:id])

  if @user.update(user_params)
    # ...
  end
end

def user_params
  params.require(:user).permit(:authority)
end
```

If you decide to start adding more roles it will avoid the complex state changes from one role to the next.